app/src/main/java/org/groundplatform/android/ui/util/RotateTransformation.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.groundplatform.android.ui.util
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Matrix
+import android.net.Uri
+import androidx.exifinterface.media.ExifInterface
+import com.squareup.picasso.Transformation
+import java.io.IOException
+
+class RotateTransformation(private val uri: Uri, private val context: Context) : Transformation {
```

Comment:
This class automatically rotates the image specifically based on Exif tags, but that's not mentioned anywhere in the class name.   Naming this something more obvious, like `RotateUsingExif` would make the type of transformation clearer.